bindings/cs/MongoDB.Crypt/LocalKeyId.cs

```diff
@@ -25,13 +27,31 @@ public class LocalKeyId : IKmsKeyId, IInternalKmsKeyId
         /// <param name="key">The key.</param>
         public LocalKeyId()
         {
+            AlternateKeyNames = new List<byte[]>().AsReadOnly();
+        }
+
+        /// <summary>
+        /// Creates an <see cref="LocalKeyId"/> class.
+        /// </summary>
+        /// <param name="alternateKeyNames">The alternate key names.</param>
+        public LocalKeyId(IEnumerable<byte[]> alternateKeyNames)
+        {
+            AlternateKeyNames = alternateKeyNames;
```

Comment:
It's dangerous to store an IEnumerable directly because we don't know whether it's deferred execution or not.

I would change the type of AlternateKeyNames to IReadOnlyList\<T> and do this:

    AlternateKeyNames = alternateKeyNames.ToList().AsReadOnly();

Reply:
Done